shell/tests/apply_and_verify_context_actions_from_file.rs

```diff
@@ -0,0 +1,194 @@
+
+    let commit_log = create_commit_log(&commit_log_db_path);
+    let kv = create_key_value_store(&key_value_db_path, &cache);
+    let storage = PersistentStorage::new(kv.clone(), kv.clone(), kv.clone(), commit_log.clone());
+    let mut context: Box<dyn ContextApi> = Box::new(TezedgeContext::new(
+        BlockStorage::new(&storage),
+        storage.merkle(),
+    ));
+
+    let block_storage = BlockStorage::new(&storage);
+
+    let actions_reader = ActionsFileReader::new(&actions_storage_path).unwrap();
+    let header = actions_reader.header();
+    info!(
+        logger,
+        "Reading info from file {}",
+        actions_storage_path.to_str().unwrap()
+    );
+    info!(logger, "{}", header);
+    let blocks_count = header.block_count;
+    let mut counter = 0;
+
+    for messages in actions_reader.take(10) {
+        counter += 1;
+        let progress = counter as f64 / blocks_count as f64 * 100.0;
+
+        match messages.iter().last() {
+            Some(msg) => match &msg.action {
+                ContextAction::Commit {
+                    block_hash: Some(block_hash),
+                    ..
+                } => {
+                    debug!(
+                        logger,
+                        "progress {:.7}% - processing block {} with {} messages",
+                        progress,
+                        hex::encode(&block_hash),
+                        messages.len()
+                    );
+                }
+                _ => {
+                    panic!("missing commit action")
+                }
+            },
+            None => {
+                panic!("missing commit action")
+            }
+        };
+
+        for msg in messages.iter() {
+            if let ContextAction::Commit {
+                block_hash: Some(block_hash),
+                ..
+            } = &msg.action
+            {
+                // there is extra validation in ContextApi::commit that verifies that
+                // applied action comes from known block - for testing purposes
+                // block_storage needs to be fed with stub value in order to pass validation
+                let mut b = header_stub.clone();
+                b.hash = BlockHash::try_from(block_hash.clone())?;
+                block_storage.put_block_header(&b).unwrap();
+            }
+
+            match &msg.action {
+                // actions that does not mutate staging area can be ommited here
+                ContextAction::Set { .. }
+                | ContextAction::Copy { .. }
+                | ContextAction::Delete { .. }
+                | ContextAction::RemoveRecursively { .. }
+                | ContextAction::Commit { .. }
+                | ContextAction::Checkout { .. } => {
+                    if let Err(e) = perform_context_action(&msg.action, &mut context) {
+                        panic!("cannot perform action {:?} error: '{}'", &msg, e);
+                    }
+                }
+                _ => {}
```

Comment:
Please, use an explicit enumeration here, we should avoid wildcard matches on enums whenever possible.